### PR TITLE
tensorrt tests ported to use VM

### PR DESCRIPTION
**Pull request by rohanmukh**
test_tensorrt.py codes modified from graph runtime to using relay VM. All the ops tested are static.

Thanks for contributing to TVM!   Please refer to guideline https://tvm.apache.org/docs/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/apache/incubator-tvm/blob/master/CONTRIBUTORS.md#reviewers) by @ them in the pull request thread.

### CODE REVIEW

**Comment by anijain2305** [comment]
@zhiics @trevor-m

File: tests/python/relay/test_tensorrt.py [hunk 1]
```diff
@@ -46,18 +47,25 @@ def test_tensorrt_simple():
...
-    mod.run(x=x_data, y=y_data, z=z_data)
-    results = [mod.get_output(i).asnumpy() for i in range(mod.get_num_outputs())]
+    mod = tvm.IRModule()
+    mod['main'] = f
+
+    result_dict = dict()
+    for mode in ['vm', 'graph']:
+        mod = relay.tensorrt.EnableTrt(mod)
+        with relay.build_config(opt_level=3):
+            ex_after = relay.create_executor(mode, mod=mod, ctx=tvm.gpu(0), target="cuda")
```
Comment (anijain2305) [inline tests/python/relay/test_tensorrt.py:60]: Minor - replace `ex_after` with better name

Comment (anijain2305) [reply tests/python/relay/test_tensorrt.py:60]: Similar applies for other instances in the file

File: tests/python/relay/test_tensorrt.py [hunk 2]
```diff
@@ -353,6 +360,7 @@ def test_conv3d_transpose(x_shape=(1, 32, 8, 8, 8), k_shape=(32, 16, 3, 3, 3), g
     run_and_verify(test_float_const())
     run_and_verify(test_multiple_outputs())
     run_and_verify(test_clip())
+
```
Comment (anijain2305) [inline tests/python/relay/test_tensorrt.py:363]: Remove

File: tests/python/relay/test_tensorrt.py [hunk 3]
```diff
@@ -460,25 +467,29 @@ def check_trt_used(graph):
...
 
-        mod = graph_runtime.create(graph, lib, ctx=tvm.gpu(0))
-        mod.set_input(**params)
         # Warmup
         for i in range(10):
-            mod.run(data=i_data)
+            if use_trt:
+                _ = exec.evaluate()(*i_data, **params)
+            else:
+                _ = exec.evaluate()(i_data, **params)
```
Comment (anijain2305) [inline tests/python/relay/test_tensorrt.py:483]: Why is this difference?

File: tests/python/relay/test_tensorrt.py [hunk 4]
```diff
@@ -523,43 +535,44 @@ def check_trt_used(graph):
         latency[model], res = test_model(model, i_data, input_shape, dtype, use_trt=True)
         _, ref_res = test_model(model, i_data, input_shape, dtype, use_trt=False, num_iteration=1)
         tvm.testing.assert_allclose(res.asnumpy(), ref_res.asnumpy(), rtol=1e-3, atol=1e-3)
-    
+
+    print('Overall runtime for models run with TRT+VM')
```
Comment (anijain2305) [inline tests/python/relay/test_tensorrt.py:539]: Remove

File: tests/python/relay/test_tensorrt.py [hunk 5]
```diff
@@ -523,43 +535,44 @@ def check_trt_used(graph):
...
         return
-    import mxnet
+
     from mxnet.gluon.model_zoo.vision import get_model
     block = get_model('resnet18_v1', pretrained=True)
+    i_data = np.random.uniform(0, 1, (1, 3, 224, 224)).astype('float32')
+
     mod, params = relay.frontend.from_mxnet(block, shape={'data': (1, 3, 224, 224)}, dtype='float32')
+
+    ## Never saved
```
Comment (anijain2305) [inline tests/python/relay/test_tensorrt.py:553]: Better comment

File: tests/python/relay/test_tensorrt.py [hunk 6]
```diff
@@ -523,43 +535,44 @@ def check_trt_used(graph):
...
-    with open('compiled.params', 'rb') as f_params:
-        params = bytearray(f_params.read())
-    lib = tvm.runtime.load_module("compiled.so")
-    # Run
-    mod = graph_runtime.create(graph, lib, ctx=tvm.gpu(0))
-    mod.load_params(params)
-    i_data = np.random.uniform(0, 1, (1, 3, 224, 224)).astype('float32')
-    for i in range(10):
-        mod.run(data=i_data)
+    code, lib = vm_exec.save()
```
Comment (anijain2305) [inline tests/python/relay/test_tensorrt.py:564]: Previously, this was testing serialize to disk and then read back from that file. I don't think we are testing ser-deser here.

**Review by anijain2305 — COMMENTED** [review]
Thanks for the PR! I have added some comments.

**Comment by rohanmukh** [comment]
Addressing the comments and reopening pull request with further changes in tensrorrt.py

**Pull request closed without merging**